discovery-provider/src/tasks/update_track_is_available.py

```diff
@@ -0,0 +1,250 @@
+                        is_available = True
+                    else:
+                        spID_replica_set = [entry[1], *entry[2]]
+                        is_available = check_track_is_available(
+                            redis=redis,
+                            track_id=track_id,
+                            spID_replica_set=spID_replica_set,
+                        )
+
+                    track_id_to_is_available_status[track_id] = is_available
+
+                # Update tracks with is_available status
+                tracks = query_tracks_by_track_ids(session, unavailable_track_ids_batch)
+                for track in tracks:
+                    is_available = track_id_to_is_available_status[track.track_id]
+
+                    # If track is not available, also flip 'is_delete' flag to True
+                    if not is_available:
+                        track.is_available = False
+                        track.is_delete = True
+
+                session.commit()
+            except Exception as e:
+                logger.warn(
+                    f"update_track_is_available.py | Could not process batch {unavailable_track_ids_batch}: {e}\nContinuing..."
+                )
+                session.rollback()
+
+
+def fetch_unavailable_track_ids(node: str) -> List[int]:
+    """Fetches unavailable tracks from Content Node. Returns empty list if request fails."""
+    unavailable_track_ids = []
+
+    try:
+        resp = requests.get(
+            f"{node}/blacklist/tracks", timeout=REQUESTS_TIMEOUT_SECONDS
+        ).json()
+        unavailable_track_ids = resp["data"]["values"]
+    except Exception as e:
+        logger.warn(
+            f"update_track_is_available.py | Could not fetch unavailable tracks from {node}: {e}"
+        )
+
+    return unavailable_track_ids
+
+
+def query_replica_set_by_track_id(
+    session: Any, track_ids: List[int]
+) -> Union[List[Tuple[int, int, List[int]]], List[Tuple[int, None, List[None]]]]:
+    """
+    Returns an array of tuples with the structure: [(track_id | primary_id | secondary_ids), ...]
+    If `primary_id` and `secondary_ids` are undefined, will return as None
+    """
+    track_ids_and_replica_sets = (
+        session.query(Track.track_id, User.primary_id, User.secondary_ids)
+        .join(User, Track.owner_id == User.user_id, isouter=True)  # left join
+        .filter(
+            User.is_current == True,
+            Track.is_current == True,
+            Track.track_id.in_(track_ids),
+        )
+        .all()
+    )
+
+    return track_ids_and_replica_sets
+
+
+def query_tracks_by_track_ids(session: Any, track_ids: List[int]) -> List[Any]:
+    """Returns a list of Track objects that has a track id in `track_ids`"""
+    tracks = (
+        session.query(Track)
+        .filter(
+            Track.is_current == True,
+            Track.track_id.in_(track_ids),
+        )
+        .all()
+    )
+
+    return tracks
+
+
+def query_registered_content_node_info(
```

Comment:
might be able to reuse `get_ursm_cnodes` instead

Reply:
ah so a few things:
1. I tried using that fn and got the following error. I think the issue was calling this fn out of the app context. I didnt want to spend to much time figuring that out, so..
```
{"levelno": 40, "level": "ERROR", "msg": "Task update_track_is_available[57b77847-f8a8-41e7-a980-9d399c669afa] raised unexpected: RuntimeError('Working outside of application context.\
\
This typically means that you attempted to use functionality that needed\
to interface with the current application object in some way. To solve\
this, set up an application context with app.app_context().  See the\
documentation for more information.')
Traceback (most recent call last):
  File \"/usr/lib/python3.9/site-packages/celery/app/trace.py\", line 385, in trace_task
    R = retval = fun(*args, **kwargs)
  File \"/usr/lib/python3.9/site-packages/celery/app/trace.py\", line 648, in __protected_call__
    return self.run(*args, **kwargs)
  File \"/audius-discovery-provider/src/tasks/update_track_is_available.py\", line 232, in update_track_is_available
    raise e
  File \"/audius-discovery-provider/src/tasks/update_track_is_available.py\", line 219, in update_track_is_available
    fetch_unavailable_track_ids_in_network(redis)
  File \"/audius-discovery-provider/src/tasks/update_track_is_available.py\", line 25, in fetch_unavailable_track_ids_in_network
    content_nodes = query_registered_content_node_info()
  File \"/audius-discovery-provider/src/tasks/update_track_is_available.py\", line 152, in query_registered_content_node_info
    registered_content_nodes = get_ursm_cnodes_endpoint_and_spID()
  File \"/audius-discovery-provider/src/queries/get_ursm_cnodes.py\", line 27, in get_ursm_cnodes_endpoint_and_spID
    db = get_db_read_replica()
  File \"/audius-discovery-provider/src/utils/db_session.py\", line 17, in get_db_read_replica
    return current_app.db_read_replica_session_manager
  File \"/usr/lib/python3.9/site-packages/werkzeug/local.py\", line 347, in __getattr__
    return getattr(self._get_current_object(), name)
  File \"/usr/lib/python3.9/site-packages/werkzeug/local.py\", line 306, in _get_current_object
    return self.__local()
  File \"/usr/lib/python3.9/site-packages/flask/globals.py\", line 51, in _find_app
    raise RuntimeError(_app_ctx_err_msg)
RuntimeError: Working outside of application context.
```
2. I wrote a custom fn that returned the fields I really needed, which was the endpoint and spID, rather than all the other info that came with SPs